### PR TITLE
Update contributors page with GitHub contributors display

### PR DESCRIPTION
**Pull request by soumyaranjan-panda**
## Contributors Page
The "contributorsPage.html" file has been created to showcases Sugar Labs' contributors with dynamically fetched GitHub API data for avatars, names, and contribution counts.

### Contents:
- **File:** `contributorsPage.html`
- **Purpose:** Display GitHub contributors on the Sugar Labs website.
- **Key Features:**
  - Asynchronously fetches contributor data using `fetchContributors.js`.
  - Styled using `contributorsPage.css` for improved visual presentation.

### Updates:
- [x] Created `contributorsPage.html` to centralize information about Sugar Labs contributors.
- [x] Updated `index.html` to include a section that redirects to Contributors Page.
- [x] Implemented JavaScript (`fetchContributors.js`) to fetch and display GitHub data.
- [x] Styled the page using `contributorsPage.css`.

### Checklist
- [x] Tested locally to verify functionality.
- [x] No merge conflicts with the latest main branch.

### Contributors Page Button Location
Contributors Page Button Location

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -494,4 +494,14 @@ <h4>170</h4>
     </div>
 </section>
 
+<!-- "Leadership page" page redirect section -->
+<section id="intro">
+    <div class="container">
+        <div class="row">
+            <div class="col-md-12 text-center">
+                <a href="{{ site.baseurl }}/leadershipPage"><button type="button" class="btn bt-raised btn-custom bt">leadership Page</button></a>
```
Comment (walterbender) [inline index.html:502]: ...>Leadership Page<...

**Comment by chimosky** [comment]
I think a page about contributors should be called contributors, although there's already a `profiles.md` which has manual entries, maybe this should replace that or be combined with that as lots of the contributors there didn't write code.

**Comment by quozl** [comment]
I agree, leadership of the organisation does not (and does not need to) correlate to code contribution.

**Comment by pikurasa** [comment]
I meant something different by https://github.com/sugarlabs/www/issues/422. Visitors to sugarlabs.org may want to know who the current leadership is, such as current management and board members.

That's what I had in mind for that ticket...

**Comment by quozl** [comment]
I agree with @pikurasa

**Comment by soumyaranjan-panda** [comment]
Walter sir, mentioned that the development community is on GitHub, so I diverted from my initial objective. 
Where can I get the current leadership data?

**Comment by quozl** [comment]
Choose whether you wish to fix the leadership issue or automate the profiles page process;

*Fix the issue*

* [ ] remove the contributors and the way they are caught from GitHub,
* [ ] convert the information from the Wiki,
* [ ] set up a way to approve changes to the Wiki,

*Automate profiles*

* [ ] remove the issue from commit messages and the opening comment of the pull request,
* [ ] integrate with the profiles page.

**Comment by soumyaranjan-panda** [comment]
@quozl  Could you please explain what I need to do for the 'Automate Profiles' part? I'm not entirely sure I understand what is required.

**Comment by soumyaranjan-panda** [comment]
Just to make sure I got it right: I need to choose between:

1. Fixing the leadership issue, which means:
   - Removing the contributors from GitHub (it might be better to create another PR without including the contributors page fetched from GitHub)
   - Converting the info from the Wiki
   - Setting up a way to approve Wiki changes (I’m not sure if this is possible, I need to do some research on it)

2. Automating the profiles page by:
   - Removing the issue from commit messages and the PR opening comment
   - Integrating it with the profiles page (the one at [Sugar Labs profiles page](https://www.sugarlabs.org/profiles/), right?)

**Comment by quozl** [comment]
> > way to approve Wiki changes

> not sure this is possible

Sugar Labs leadership have legal recognition and responsibility for the organisation, and some are probably _listed_ in documents presented to the US government, yet the way in which the leadership is _listed_ on the web sites is a page on the Wiki, which can be edited by anybody who registers for the Wiki.  It's like having a list of doctors in a clinic written on a whiteboard at the front door, _and_ providing a marker and eraser.  It is more common for a list like this to be set in stone or metal.  You can get closer to that ideal by adding an approval step.  Passive approval, of hoping someone notices a change, is not nearly so ideal.

This isn't only an HTML5 type of problem, but a governance and organisational problem.  The people it has to involve are the leadership.  They probably aren't all listening here, so you can't assume that any proposal here has their agreement unless they all say so in response to your proposal.  Tabling it at a meeting is one way.  Officers of the organisation can also take decisive action to make changes, and report them to the next meeting.

https://wiki.sugarlabs.org/go/Oversight_Board lists the oversight board members and the officers.  I'm no longer in either, so what I say is only a suggestion.

There are many ways to arrange the approval method.  If all the leaders can use GitHub (not something I've been able to verify!), then a submodule repository with access controls that holds the list may be one way.  The Wiki page could be protected.

> the one at Sugar Labs profiles page, right?

Yes.  Retaining the existing list of names and adding the newer names of GitHub contributors.

> should I go for it?

I'm puzzled by your question.  Up to you.  There are advantages and disadvantages.

Advantages;

* Sugar Labs properly describes leadership on the site,
* a process is set up for change when the leadership changes (as a result of a meeting),
* list of contributors is more up to date than now, giving social credit,

Disadvantages;

* your time,
* you are displaced from working on Sugar Labs software; Sugarizer, Music Blocks, Sugar, and activities,
* not much of a feather in your cap, as so few people care for the web site; it is a necessary but very very small part of Sugar Labs work.

Hope that helps!

**Comment by pikurasa** [comment]
FWIW, I locked the wiki page.

Basically, the names, headshots, and titles of the people on https://wiki.sugarlabs.org/go/Oversight_Board would be added to a new "leadership" page.

**Comment by quozl** [comment]
Nice.

**Comment by pikurasa** [comment]
For anyone wondering what this contributor page looks like, it looks like this:

Screenshot from 2024-10-01 15-12-00

**Comment by pikurasa** [comment]
@soumyaranjan-panda I don't mind adding this contributor page, but, if we were to add this page, I'd want to have documentation within this repo for how to maintain this page, how to update it.

But, this is also different than a leadership page.

**Comment by soumyaranjan-panda** [comment]
> @soumyaranjan-panda I don't mind adding this contributor page, but, if we were to add this page, I'd want to have documentation within this repo for how to maintain this page, how to update it.
> 
> But, this is also different than a leadership page.

Got it, @pikurasa . I'll make sure to document it.
I've been a little busy with my academics, but once I'm free, I'll work on the leadership page.

**Comment by soumyaranjan-panda** [comment]
@pikurasa We don't need to manually update this page. It will be updated dynamically. What I’ve implemented is that every time someone visits the contributors' page, it fetches the latest data from GitHub and updates the information accordingly.
Do you still need documentation for this?

**Comment by quozl** [comment]
But doesn't that make our web site critically depend on github.com instead of sugarlabs.org alone?  That doesn't seem safe.

**Comment by soumyaranjan-panda** [comment]
How about we manually fetch the data and update it periodically instead? This way, we can maintain control and reduce our dependency on GitHub, ensuring that our website remains secure.

**Pull request closed without merging**